### PR TITLE
Docs: Expanding on the answers in the FAQ

### PR DESCRIPTION
**Pull request by Exempt-Medic**
## What is this fixing or adding?

Adds a bit of text explaining the value in `early_local_items`, adds alternative ways to handle the restrictive start fill errors, and adds alternatives to the `while` loop for creating filler items.

## How was this tested?

Reading

### CODE REVIEW

File: docs/apworld_dev_faq.md [hunk 1]
```diff
@@ -6,12 +6,18 @@ This document is meant as a reference tool to show solutions to common problems
 
 ### My game has a restrictive start that leads to fill errors
 
-Hint to the Generator that an item needs to be in sphere one with local_early_items
+Hint to the Generator that an item needs to be in sphere one with local_early_items. Here `1` represents the number of "Sword" items to attempt to place in sphere 1.
```
[the remark this answers is not in the document]
Comment (Exempt-Medic) [reply docs/apworld_dev_faq.md:9]: I even fixed one instance of it earlier 🤦

**Comment by qwint** [comment]
if `while` and `for _ in range` are equivalent but while could potentially run into issues (which is true afaik) i'd rather just replace the `while` version
the list comprehension as an alternative looks fine though

**Comment by Exempt-Medic** [comment]
> if `while` and `for _ in range` are equivalent but while could potentially run into issues (which is true afaik) i'd rather just replace the `while` version the list comprehension as an alternative looks fine though

I was more just worried that the `for` format was less intuitive since it uses `_`

**Review by qwint — APPROVED** [review]

File: docs/apworld_dev_faq.md [hunk 2]
```diff
@@ -6,12 +6,18 @@ This document is meant as a reference tool to show solutions to common problems
 
 ### My game has a restrictive start that leads to fill errors
 
-Hint to the Generator that an item needs to be in sphere one with local_early_items
+Hint to the Generator that an item needs to be in sphere one with local_early_items. Here `1` represents the number of "Sword" items to attempt to place in sphere one.
```
Comment (ScipioWright) [inline docs/apworld_dev_faq.md:9]:
```suggestion
Hint to the Generator that an item needs to be in sphere one with local_early_items. Here, `1` represents the number of "Sword" items to attempt to place in sphere one.
```

**Review by ScipioWright — APPROVED** [review]

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**